build_tools/python/e2e_test_framework/definitions/utils.py

```diff
@@ -0,0 +1,50 @@
+## Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Utils that help construct definitions."""
+
+from typing import Callable, List, Sequence
+
+MAX_SUBSTITUTION_ITERATIONS = 10
+
+
+def transform_flags(flags: Sequence[str],
```

Comment:
transform_flags is added to allow us to uniformly do flag value substitution/transformation. See https://github.com/openxla/iree/pull/12757/files#diff-5abdbb3b5dbffcc6b64b9b2918f5713b4960c6b36b6fac5701940f453cea8990R277 as an example